source/dub/project.d

```diff
@@ -1542,6 +1550,18 @@ final class SelectedVersions {
 		m_dirty = true;
 	}
 
+	/// Selects a certain Git reference for a specific package.
+	void selectVersion(string package_id, Repository repository, string spec)
```

Comment:
Please pick a more descriptive method name such as `selectVersionWithRepository` or `selectGitVersion`. dub.d:591 is extremely difficult to read because it is an overload of selectVersion using repository + versionSpec instead of version_ which is already confusing enough.

Stating it another time here in the method name that this is for git would make sense imo to make the overload clearer.